sqlite-db/src/insert_rollover_completed_event_data.rs

```diff
@@ -0,0 +1,304 @@
+use anyhow::Result;
+use bdk::bitcoin::hashes::hex::ToHex;
+use model::Cet;
+use model::CfdEvent;
+use model::Dlc;
+use model::EventKind;
+use model::FundingFee;
+use model::OrderId;
+use model::RevokedCommit;
+use sqlx::pool::PoolConnection;
+use sqlx::Connection as SqlxConnection;
+use sqlx::Sqlite;
+use sqlx::Transaction;
+
+pub async fn insert_rollover_completed_event(
+    connection: &mut PoolConnection<Sqlite>,
+    event_id: i64,
+    event: CfdEvent,
+) -> Result<()> {
+    let event_kind = event.event;
+    match event_kind {
+        EventKind::RolloverCompleted {
+            dlc: Some(dlc),
+            funding_fee,
+        } => {
+            // we begin a transaction so that it rollsback in case of a failure further down
+            let mut inner_transaction = connection.begin().await?;
+
+            insert_rollover_completed_event_data(
+                &mut inner_transaction,
+                event_id,
+                &dlc,
+                funding_fee,
+                event.id,
+            )
+            .await?;
+
+            for revoked in dlc.revoked_commit {
+                insert_revoked_commit_transaction(&mut inner_transaction, event.id, revoked)
+                    .await?;
+            }
+
+            for (_event_id, cets) in dlc.cets {
+                // ignoring event_id for now because we only have one event
+                for cet in cets {
+                    insert_cet(&mut inner_transaction, event.id, cet).await?;
+                }
+            }
+
+            // Commit the transaction to either write all or rollback
+            inner_transaction.commit().await?;
+        }
+        EventKind::RolloverCompleted { dlc: None, .. } => {
+            // We ignore rollover completed events without DLC data as we don't need to store
+            // anything
+        }
+        _ => {
+            tracing::error!("Invalid event type. Use `append_event` function instead")
+        }
+    }
+
+    Ok(())
+}
+
+/// Inserts RolloverCompleted data and returns the resulting rowid
+async fn insert_rollover_completed_event_data(
+    inner_transaction: &mut Transaction<'_, Sqlite>,
+    event_id: i64,
+    dlc: &Dlc,
+    funding_fee: FundingFee,
+    offer_id: OrderId,
+) -> Result<()> {
+    let (lock_tx, lock_tx_descriptor) = dlc.lock.clone();
+    let (commit_tx, commit_adaptor_signature, commit_descriptor) = dlc.commit.clone();
+    let (refund_tx, refund_signature) = dlc.refund.clone();
+
+    // casting because u64 is not implemented for sqlx: https://github.com/launchbadge/sqlx/pull/919#discussion_r557256333
+    let funding_fee_as_sat = funding_fee.fee.as_sat() as i64;
+    // TODO: these seem to be redundant and should be in `cfds` table only
```

Comment:
Why do we need this information in the rolloverevent?

Reply:
It's a fair question because rollover happens on top of lock. But it's just convenient to build the `Dlc` based on one event. Otherwise I think aggregates will get more complicated.

Reply:
Until we get rid of the aggregates (😈) we could do a sql join and load this information from the other table. That way we don't store reduntant information and would not have to change the model.